### ISSUE TITLE
Delete a file

### ISSUE THREAD

**Issue by asecondwill** [josegonzalez/cakephp-upload#361]
How is deleting a file for a record handled? 

So if I'm editing a user and he deletes his profile pic, eg by ticking a checkbox to delete it, is there a method on the model / behaviour to clear the image from the db record and delete the image?

**Comment by josegonzalez** [comment]
In the 3.x version, files are not deleted automatically.

**Comment by asecondwill** [comment]
any plans to add that?

**Comment by asecondwill** [comment]
also, what about clearing the actual db records? literally just by hand: 

```
 $record->image = ""; 
 $record->image_size = 0; 
 $record->image_dir = ""; 
 $record->image_type = ""; 
// save
```

**Comment by josegonzalez** [comment]
I literally do none of that at the moment. I'm not doing active development on this plugin at this time - I really only touch it if I'm bored - but I would definitely pull in a PR that adds that functionality.

The one niggle you'd have to worry about is generating all the filenames upon deletion. It should be possible, but in theory users can also setup their FS such that the files are not deletable for whatever reason.

**Comment by jorisvaesen** [comment]
@asecondwill we're working on it: https://github.com/josegonzalez/cakephp-upload/pull/362

**Comment by asecondwill** [comment]
@jorisvaesen  - thanks.  Good to hear it's being maintained.

**Comment by jorisvaesen** [comment]
@asecondwill This feature is available with the release of 3.3.0.
